pkg/proxy/endpointslicecache.go

```diff
@@ -108,11 +108,17 @@ func newEndpointSliceTracker() *endpointSliceTracker {
 // newEndpointSliceInfo generates endpointSliceInfo from an EndpointSlice.
 func newEndpointSliceInfo(endpointSlice *discovery.EndpointSlice, remove bool) *endpointSliceInfo {
 	esInfo := &endpointSliceInfo{
-		Ports:     endpointSlice.Ports,
+		Ports:     make([]discovery.EndpointPort, len(endpointSlice.Ports), len(endpointSlice.Ports)),
 		Endpoints: []*endpointInfo{},
 		Remove:    remove,
 	}
 
+	// ports are sorted for diffing later, deep copy required here to avoid
```

Comment:
do we need to deep copy the ports, or just copy the slice? do we actually modify the port items or just the order?

Reply:
if the latter, we could just do:

```go
Ports: make([]discovery.EndpointPort, len(endpointSlice.Ports)),
...
copy(esInfo.Ports, endpointSlice.Ports)
```

Reply:
Good catch, thanks! Updated to just do the copy.